### PR TITLE
[multistage] support the rest of the agg functions currently implementable

### PR DESCRIPTION
**Pull request by walterddr**
add all supported agg functions to follow up with #11034 

the following can be added very soon (some complication due to testing)
- `distinctCountBitmap` is not yet supported, can be added later 
- all `*TupleSketch` functions not yet supported
- `STUnion` agg function 
- all `*MV` variance of agg functions will be added in #11216

the following functions require some revisit
- `histogram` not yet supported due to complex ARRAY constructor
- `_` version of the stats function such as `COVAR_POP` is not supported, (where as `coVarPop` is) due to calctie built-in operator table conflict.
- `argmin/argmax` is not supported due to conflicting signature (e.g. arg column at 1st vs arg column at last convention differs between pinot and calcite)
- `funnelCount` due to its complex syntax, there should be an easier solution in v2

### CODE REVIEW

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java [hunk 1]
```diff
@@ -86,23 +87,32 @@ public enum AggregationFunctionType {
   DISTINCTCOUNTHLL("distinctCountHLL", ImmutableList.of("DISTINCT_COUNT_HLL"), SqlKind.OTHER_FUNCTION,
       SqlFunctionCategory.USER_DEFINED_FUNCTION, OperandTypes.ANY, ReturnTypes.BIGINT,
       ReturnTypes.explicit(SqlTypeName.OTHER)),
-  DISTINCTCOUNTRAWHLL("distinctCountRawHLL"),
-  DISTINCTCOUNTSMARTHLL("distinctCountSmartHLL"),
+  DISTINCTCOUNTRAWHLL("distinctCountRawHLL", ImmutableList.of("DISTINCT_COUNT_RAW_HLL"), SqlKind.OTHER_FUNCTION,
```
Comment (walterddr) [inline pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java:90]: test needed for optional 2nd argument about HLL log2m size

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java [hunk 2]
```diff
@@ -86,23 +87,32 @@ public enum AggregationFunctionType {
...
+      SqlFunctionCategory.USER_DEFINED_FUNCTION,
+      OperandTypes.family(ImmutableList.of(SqlTypeFamily.ANY, SqlTypeFamily.INTEGER), ordinal -> ordinal > 0),
+      ReturnTypes.VARCHAR_2000, ReturnTypes.explicit(SqlTypeName.OTHER)),
+  DISTINCTCOUNTSMARTHLL("distinctCountSmartHLL", ImmutableList.of("DISTINCT_COUNT_SMART_HLL"), SqlKind.OTHER_FUNCTION,
```
Comment (walterddr) [inline pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java:94]: test needed for optional 2nd argument about HLL parameter string

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java [hunk 3]
```diff
@@ -138,17 +148,23 @@ public enum AggregationFunctionType {
   FOURTHMOMENT("fourthMoment"),
 
   // DataSketches Tuple Sketch support
-  DISTINCTCOUNTTUPLESKETCH("distinctCountTupleSketch"),
+  DISTINCTCOUNTTUPLESKETCH("distinctCountTupleSketch", null,
```
Comment (walterddr) [inline pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java:151]: this is not supported yet. should removed and move the TODO above

File: pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/LiteralHintUtils.java [hunk 4]
```diff
@@ -46,13 +46,13 @@ public static String literalMapToHintString(Map<Pair<Integer, Integer>, RexExpre
           e.getValue().getDataType().name(), e.getValue().getValue()));
     }
     // semi-colon is used to separate between encoded literals
-    return "{" + StringUtils.join(literalStrings, ";") + "}";
+    return "{" + StringUtils.join(literalStrings, ";:;") + "}";
```
Comment (xiangfu0) [inline pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/LiteralHintUtils.java:49]: where is this `;:;` used?

**Review by xiangfu0 — APPROVED** [review]

File: pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/LiteralHintUtils.java (hunk 4 above)
Comment (walterddr) [reply pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/LiteralHintUtils.java:49]: the problem is b/c in some agg literals it allows stringify parameter type with `;` as the separator. this conflicts with my usage of `;` which indicates the "next literal hint".

Thus i used another 3-character splitter `;:;` to indicate hint-literal boundaries.

**Pull request merged**